pkg/types/chain_writer.go

```diff
@@ -7,12 +7,94 @@ import (
 	"github.com/google/uuid"
 )
 
+// CallOption can be used to configure a single call to the ChainWriter.
+type CallOption interface {
+	apply(*CallOptions)
+}
+
+// CallOptionFunc is a functional type that implements the CallOption interface.
+type callOptionFunc func(*CallOptions)
+
+func (fn callOptionFunc) apply(o *CallOptions) {
+	fn(o)
+}
+
+// CallOptions contains the values which are configurable per-call to the ChainWriter.
+//
+// ChainWriter implementations should hold an instance of this struct and apply the options to the struct before making the call.
+type CallOptions struct {
+	// TransactionID is submitted to the particular chain's TXM as an idempotency key.
+	// It should be unique across transaction.
+	TransactionID   string
+	TransactionMeta *TxMeta
+	Value           *big.Int
+}
+
+// Apply applies the given options to the CallOptions.
+func (o *CallOptions) Apply(opts ...CallOption) {
+	for _, opt := range opts {
+		opt.apply(o)
+	}
+}
+
+// WithTransactionID sets the transactionID (idempotency key) for a call to the ChainWriter.
+//
+// Default: ""
+func WithTransactionID(transactionID string) CallOption {
+	return callOptionFunc(func(o *CallOptions) {
+		o.TransactionID = transactionID
```

Comment:
If we do make this optional, we should probably create a new uuid and convert it to a string to ensure uniqueness.

Reply:
Actually, I guess the TXM can just take empty string and deal with it appropriately, so maybe it shouldn't have an ID assigned. It just means that you can never check the Status() of it, even if you wanted to later.

Reply:
I think the `TXM` already handles an empty string, but we can create a random UUID as the default too.